model/src/application/interaction/message_component/mod.rs

```diff
@@ -49,6 +49,16 @@ pub struct MessageComponentInteraction {
     pub user: Option<User>,
 }
 
+impl MessageComponentInteraction {
+    /// Id of the user that invoked this interaction
+    pub fn author_id(&self) -> Option<UserId> {
```

Comment:
Shouldn't this always return a `UserId`?
Something like this:
```rust
    pub fn author_id(&self) -> UserId {
        match (self.member.as_ref(), self.user.as_ref()) {
            (None, Some(u)) => u.id,
            (Some(m), None) => m.user.as_ref().expect("user is set").id,
            _ => unreachable!("member or user is always set"),
        }
```

Reply:
Yeah, it should ideally. However, I'm not sure about the policy on adding unreachable branches in the code

Reply:
Also, should discord add any other way in which member & user both become `None` (don't ask me what), this might panic (unsure?)

Reply:
I'm not against `unreachable!`s when it comes to our own APIs, but we have a policy of validating all data that comes from Discord even when its public API documentation states that this information should always be available in the two separate cases. In this case `Option<UserId>` is correct in my opinion.